server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java

```diff
@@ -98,55 +98,59 @@ protected void masterOperation(Task task, final IndicesAliasesRequest request, f
                 String[] concreteIndices =
                     indexNameExpressionResolver.concreteIndexNames(state, request.indicesOptions(), true, action.indices());
                 List<String> nonBackingIndices = Arrays.stream(concreteIndices)
                     .map(resolvedIndex -> state.metadata().getIndicesLookup().get(resolvedIndex))
                     .filter(ia -> ia.getParentDataStream() == null)
                     .map(IndexAbstraction::getName)
                     .collect(Collectors.toList());
-                if (nonBackingIndices.isEmpty() == false) {
-                    throw new IllegalArgumentException("expressions " + Arrays.toString(action.indices()) +
-                        " that match with both data streams and regular indices are disallowed");
-                }
-
                 switch (action.actionType()) {
                     case ADD:
+                        // Fail if parameters are used that data stream aliases don't support:
+                        if (action.filter() != null) {
+                            throw new IllegalArgumentException("aliases that point to data streams don't support filters");
+                        }
+                        if (action.routing() != null) {
+                            throw new IllegalArgumentException("aliases that point to data streams don't support routing");
+                        }
+                        if (action.indexRouting() != null) {
+                            throw new IllegalArgumentException("aliases that point to data streams don't support index_routing");
+                        }
+                        if (action.searchRouting() != null) {
+                            throw new IllegalArgumentException("aliases that point to data streams don't support search_routing");
+                        }
+                        if (action.isHidden() != null) {
+                            throw new IllegalArgumentException("aliases that point to data streams don't support is_hidden");
+                        }
+                        // Fail if expressions match both data streams and regular indices:
+                        if (nonBackingIndices.isEmpty() == false) {
+                            throw new IllegalArgumentException("expressions " + Arrays.toString(action.indices()) +
+                                " that match with both data streams and regular indices are disallowed");
+                        }
                         for (String dataStreamName : concreteDataStreams) {
                             for (String alias : concreteDataStreamAliases(action, state.metadata(), dataStreamName)) {
                                 finalActions.add(new AliasAction.AddDataStreamAlias(alias, dataStreamName, action.writeIndex()));
                             }
                         }
-                        break;
+                        continue;
```

Comment:
Does the `continue` here do something different than the `break` would have done?

Reply:
In this case data stream aliases are added, if `break` were to be used here, then this transport action could attempt adding indices aliases with the same name (which would then fail). A few lines below, at the end of the if statement a `continue` statement used to do what this `continue` statement is doing.